contracts/PoaOperator.sol

```diff
@@ -167,7 +167,12 @@ contract PoaOperator is Adminable {
       // if slot not active, prevent submission
       require(lastCompleteEpoch + 2 < slot.activationEpoch);
     }
-    bytes32 periodRood = keccak256(_blocksRoot, _slotId << 160 | uint160(msg.sender));
+    bytes32 periodRood = bytes32(_slotId << 160 | uint160(msg.sender));
```

Comment:
replace with `keccak256(abi.encodePacked(_blocksRoot, _slotId << 160 | uint160(msg.sender)))` instead of using assembly?

Reply:
abi.encoded is super gas inefficient. to be avoided

Reply:
https://twitter.com/nicksdjohnson/status/1044904764146962432